lib/mapping/MapFormatJson.cpp

```diff
@@ -1298,7 +1293,7 @@ std::string CMapSaverJson::writeTerrainTile(const TerrainTile & tile)
 	out.setf(std::ios::dec, std::ios::basefield);
 	out.unsetf(std::ios::showbase);
 
-	out << terrainCodes.at(int(tile.terType)) << (int)tile.terView << flipCodes[tile.extTileFlags % 4];
+	out << tile.terType.toString().substr(0, 2) << (int)tile.terView << flipCodes[tile.extTileFlags % 4];

```

Comment:
Terrain code is not the first 2 letters of terrain name. See `TerrainDetail::terrainCodes`

Reply:
@alexvins , I have no way to debug this code. Please, help with obtaining maps in json format

Reply:
Have you run the test suite?

Reply:
@alexvins in order to support new terrains everywhere we have to patch the redactor as well. Could you help with it in a later phase when we have json format stable? Or should I consider supporting this activity?

Reply:
Our redactor is only the way to create a new map with new terrains. Otherwise we can relay only on autogenerated maps

Reply:
> Our redactor is only the way to create a new map with new terrains. Otherwise we can relay only on autogenerated maps

I mean test suite covers map generation, saving to json and loading back It should be enough to test/debug this code

Reply:
Yes but this does not solve the problem that there is no tool to create an authored map with new terrains. Also later new towns might start using new terrains in their configs and so on so the redactor patch is inevitable at the end.

Reply:
Probably it is too early to consider this activity so. There is a lot of other work and now we can just use the test suit

Reply:
I will try to build vcmi_editor from source so and consider to publish a new release binaries for windows

Reply:
@alexvins , do you mean unit tests? If so, unfortunately they are broken even on develop branch.
https://h3vcmi.slack.com/archives/C2J9QFUDS/p1652881185132249

Reply:
What is sense to keep functionality which is not used and not supported? If it used, some maps should be available somewhere:)

Reply:
Might be that modders are not interested in creating custom authored maps for VCMI only?

Reply:
Right, probably, there should be two-way converter

Reply:
Two-way not possible. H3M can not handle our new objects. One way exists, I've sent you a few maps to slack

Reply:
Custom map forces users to use certain set mods, while people like to select the mods they use themselves. This is probably why many people may prefer random maps with their own rules/mods. Anyway editor and map format is of course needed and is a must have, it is just that the only way for it to be in real use is if VCMI gets some big mod.   Probably people from MDT team had some plans to do custom maps/campaigns, though it was never published or continued as far as I know.